Bug 1102506 - labelling buttons with aria

jobara opened the pull request:
Rather than having the text as invisible text nodes. The labels for the sync and settings buttons in the calendar settings are now set using an aria-label.

https://bugzilla.mozilla.org/show_bug.cgi?id=1102506

try-server-hook commented:
[<img src="https://avatars.githubusercontent.com/u/556024?v=3" alt="jobara" width="50px" height="50px" /> Justin Obara (jobara)](https://github.com/jobara) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=3d9b63de170d)

yzen commented on apps/calendar/locales/calendar.en-US.properties:
```diff
@@ -32,7 +32,9 @@ account-header=Account
 # advanced settings
-advanced-settings-short=Settings
+# This settings button uses an icon for display, so this text is only
+# for accessibility purposes, not visible to the eye.
```
Use the default comment for screen reader strings: https://github.com/jobara/gaia/blob/1102506/apps/calendar/locales/calendar.en-US.properties#L85-L86

yzen commented on apps/calendar/elements/settings.html:
```diff
@@ -14,11 +14,11 @@
           <button class="settings toolbar-item">
             <span class="icon"
-                  data-l10n-id="advanced-settings-short">Settings</span>
+                  data-l10n-id="advanced-settings-short"></span>
```
When updating strings we must change the key (rule imposed by l10n team) so for the two that you are changing lets have: advanced-settings-short-icon-button and drawer-sync-icon-button

yzen commented:
Also I would check if the font size set to 0 CSS style for those icons is now redundant (if it was purposefully added).

try-server-hook commented:
[<img src="https://avatars.githubusercontent.com/u/556024?v=3" alt="jobara" width="50px" height="50px" /> Justin Obara (jobara)](https://github.com/jobara) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=38c23435d792)

jobara commented:
@yzen  I'm not sure exactly where you saw the 0 CSS style. I saw something for a back button which doesn't appear to exist in the UI anymore. I've removed this with the rest of the updates.

yzen commented on apps/calendar/locales/calendar.en-US.properties:
```diff
@@ -32,7 +32,9 @@ account-header=Account
-advanced-settings-short=Settings
+# The following string is spoken by screen readers and not shown on the
+# screen.
+advanced-settings-short-icon-button.ariaLabel=Settings
```
Could you group the 2 strings together and change string to strings and is to are?

jobara replied:
Do you mean to group together the strings from line 37 and line 82?

yzen replied:
Yes

yzen commented on apps/calendar/style/settings.css:
```diff
@@ -32,14 +32,6 @@ body[data-path^='/settings'].pending-operation #progress-indicator {
-The text in the back button is just for accessibility concerns,
-not actually shown since the top header should be transparent.
-*/
-#settings .settings-back {
```
Yeah that's the type of thing i was hoping to remove, but this is related to a different button. It can be a follow up if you'd like to open a bug, but we should not remove this stuff in this PR

yzen commented on apps/calendar/test/unit/views/settings_test.js:
```diff
@@ -67,10 +67,6 @@ suiteGroup('views/settings', function() {
     div.id = 'test';
     div.innerHTML = [
       '<div id="settings">',
-      '  <header>',
```
This should not be removed?

jobara committed:
Bug 1102506 - labelling buttons with aria

Rather than having the text as invisible text nodes. The labels for the sync and settings buttons in the calendar settings are now set using an aria-label.

jobara commented:
@yzen i put back in those parts relating to the back button.

try-server-hook commented:
[<img src="https://avatars.githubusercontent.com/u/556024?v=3" alt="jobara" width="50px" height="50px" /> Justin Obara (jobara)](https://github.com/jobara) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=b10d40d16424)

try-server-hook commented:
[<img src="https://avatars.githubusercontent.com/u/556024?v=3" alt="jobara" width="50px" height="50px" /> Justin Obara (jobara)](https://github.com/jobara) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=374acdc5e026)

yzen commented:
^ re-triggering tests

try-server-hook commented:
[<img src="https://avatars.githubusercontent.com/u/561275?v=3" alt="yzen" width="50px" height="50px" /> Yura Zenevich (yzen)](https://github.com/yzen) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=cd045077a53d)

try-server-hook commented:
[<img src="https://avatars.githubusercontent.com/u/561275?v=3" alt="yzen" width="50px" height="50px" /> Yura Zenevich (yzen)](https://github.com/yzen) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=9fdf5ee48301)

try-server-hook commented:
[<img src="https://avatars.githubusercontent.com/u/556024?v=3" alt="jobara" width="50px" height="50px" /> Justin Obara (jobara)](https://github.com/jobara) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=4a96348bf141)

try-server-hook commented:
[<img src="https://avatars.githubusercontent.com/u/556024?v=3" alt="jobara" width="50px" height="50px" /> Justin Obara (jobara)](https://github.com/jobara) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=2721c704516b)
